packages/mobile/src/screens/edit-track-screen/EditTrackScreen.tsx

```diff
@@ -227,39 +229,48 @@ const getInitialBpm = (bpm: number | null | undefined) => {
 
 export const EditTrackScreen = (props: EditTrackScreenProps) => {
   const editTrackSchema = toFormikValidationSchema(useEditTrackSchema())
+  const { track } = useContext(UploadFileContext)
 
   const { initialValues: initialValuesProp, onSubmit, ...screenProps } = props
 
-  const initialValues: FormValues = {
-    ...initialValuesProp,
-    entityType: 'track',
-    licenseType: computeLicenseVariables(
-      initialValuesProp.license || ALL_RIGHTS_RESERVED_TYPE
-    ),
-    musical_key: initialValuesProp.musical_key
-      ? parseMusicalKey(initialValuesProp.musical_key)
-      : undefined,
-    bpm: getInitialBpm(initialValuesProp.bpm)
-  }
+  const initialValues: FormValues = useMemo(() => {
+    const commonProps = {
+      licenseType: computeLicenseVariables(
+        initialValuesProp.license || ALL_RIGHTS_RESERVED_TYPE
+      ),
+      musical_key: initialValuesProp.musical_key
+        ? parseMusicalKey(initialValuesProp.musical_key)
+        : undefined,
+      bpm: getInitialBpm(initialValuesProp.bpm)
+    }
+
+    if (initialValuesProp.isUpload && track) {
+      return {
+        ...track.metadata,
+        isUpload: initialValuesProp.isUpload,
+        entityType: 'track',
+        ...commonProps
+      }
+    } else {
+      return { ...initialValuesProp, entityType: 'track', ...commonProps }
+    }
+  }, [initialValuesProp, track])
 
   const handleSubmit = useCallback(
     (values: FormValues, { setSubmitting }) => {
-      const metadata = getUploadMetadataFromFormValues(
-        values,
-        initialValuesProp
-      )
+      const metadata = getUploadMetadataFromFormValues(values, initialValues)
 
       // submit the metadata
       onSubmit(metadata)
       setSubmitting(false)
     },
-    [initialValuesProp, onSubmit]
+    [initialValues, onSubmit]
   )
 
   return (
     <Formik<FormValues>
       initialValues={initialValues}
-      enableReinitialize
+      enableReinitialize={!initialValuesProp.isUpload}
```

Comment:
just checking that replace track will still work properly on the edit page?

Reply:
yea this won't change anything for the edit page, it will still be true